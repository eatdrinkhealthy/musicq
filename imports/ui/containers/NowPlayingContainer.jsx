// @flow
import React, {
  Component,
} from "react";
import Player from "../components/Player";
import RequestsContainer from "./RequestsContainer";

type IState = {
  nowPlayingUrl: string,
};

class NowPlayingContainer extends Component {
  constructor() {
    super();
    this.state = {
      nowPlayingUrl: "https://www.youtube.com/watch?v=uUcEGOLfUTE",
    };
  }

  state: IState;

  handleSelectRequest(nextRequestUrl: string) {
    this.setState({ nowPlayingUrl: nextRequestUrl });
  }

  render() {  // eslint-disable-line flowtype/require-return-type
    return (
      <div>
        <br />
        <div className="row">
          <div className="eight columns offset-by-one">
            <Player nowPlayingUrl={this.state.nowPlayingUrl} />
          </div>
        </div>
        <div className="row">
          <div className="eight columns offset-by-one">
            <RequestsContainer
              handleSelectRequest={(snr: string): void => this.handleSelectRequest(snr)}
            />
          </div>
        </div>
      </div>
    );
  }
}

export default NowPlayingContainer;
