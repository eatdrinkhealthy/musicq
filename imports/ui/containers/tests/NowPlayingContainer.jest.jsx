// @flow
/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import renderer from "react-test-renderer";
import NowPlayingContainer from "../NowPlayingContainer";

describe("<NowPlayingContainer />", function () {
  it("matches rendered snapshot", function () {
    const tree = renderer.create(<NowPlayingContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
