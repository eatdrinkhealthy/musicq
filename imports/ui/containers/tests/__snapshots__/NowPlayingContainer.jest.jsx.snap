exports[`<NowPlayingContainer /> matches rendered snapshot 1`] = `
<div>
  <br />
  <div
    className="row">
    <div
      className="eight columns offset-by-one">
      <div
        className="flex-video player">
        <div
          className={undefined}
          hidden={false}
          style={
            Object {
              "height": 360,
              "width": 640,
            }
          }>
          <div
            style={
              Object {
                "display": "block",
                "height": "100%",
                "width": "100%",
              }
            }>
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row">
    <div
      className="eight columns offset-by-one">
      <div>
        <div>
          <h4>
            Requests
          </h4>
          <ul>
            <div
              onClick={[Function]}>
              <span>
                Duke Ellington
                 - 
              </span>
              <span>
                Jeep\'s Blues
              </span>
            </div>
            <div
              onClick={[Function]}>
              <span>
                The Mustangs
                 - 
              </span>
              <span>
                Ya Mar
              </span>
            </div>
            <div
              onClick={[Function]}>
              <span>
                The Black Crowes
                 - 
              </span>
              <span>
                Gone
              </span>
            </div>
            <div
              onClick={[Function]}>
              <span>
                Grateful Dead
                 - 
              </span>
              <span>
                Franklin\'s Tower
              </span>
            </div>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
